src/Hamiltonians/HOCartesianCentralImpurity.jl

```diff
@@ -0,0 +1,200 @@
+potential at the centre, with strength `g`,
+```math
+\\hat{H}_\\mathrm{rel} = \\sum_\\mathbf{i} ϵ_\\mathbf{i} n_\\mathbf{i} 
+    + g\\sum_\\mathbf{ij} V_\\mathbf{ij} a^†_\\mathbf{i} a_\\mathbf{j}.
+```
+For a ``D``-dimensional harmonic oscillator indices ``\\mathbf{i}, \\mathbf{j}, \\ldots`` 
+are ``D``-tuples. The energy scale is defined by the first dimension i.e. ``\\hbar \\omega_x`` 
+so that single particle energies are 
+```math
+    \\frac{\\epsilon_\\mathbf{i}}{\\hbar \\omega_x} = (i_x + 1/2) + \\eta_y (i_y+1/2) + \\ldots.
+```
+The factors ``\\eta_y, \\ldots`` allow for anisotropic trapping geometries and are assumed to 
+be greater than `1` so that ``\\omega_x`` is the smallest trapping frequency.
+
+Matrix elements ``V_{\\mathbf{ij}}`` are for a delta function potential calculated in this basis
+```math
+    V_{\\mathbf{ij}} = \\prod_{d \\in x, y,\\ldots} \\psi_{i_d}(0) \\psi_{j_d}(0).
+```
+Only even parity states feel this impurity, so all ``i_d`` are even. Note that the matrix 
+representation of this Hamiltonian is completely dense in the even-parity subspace.
+
+# Arguments
+
+* `addr`: the starting address, defines number of particles and total number of modes.
+* `M=num_modes(addr) - 1`: the maximum harmonic oscillator index number in the ``x``-dimension.
+    Must be even. Index number for the harmonic oscillator groundstate is `0`.
+* `ηs=()`: a tuple of aspect ratios for the remaining dimensions `(η_y, ...)`. Should be empty 
+    for a 1D trap or contain values greater than `1.0`. The maximum index 
+    in other dimensions will be the largest even number less than `M/η_y`.
+* `g=1.0`: the strength of the delta impurity in (``x``-dimension) trap units.
+* `impurity_only=false`: if set to `true` then the trap energy terms are ignored. Useful if 
+    only energy shifts due to the impurity are required.
+
+!!! warning
+        Due to use of `SpecialFunctions` with large arguments the matrix representation of 
+        this Hamiltonian may not be strictly symmetric, but is approximately symmetric within
+        machine precision.
+
+See also [`HOCartesianContactInteractions`](@ref) and[`HOCartesianEnergyConservedPerDim`](@ref).
+"""
+struct HOCartesianCentralImpurity{
+    D,  # number of dimensions
+    A
+} <: AbstractHamiltonian{Float64}
+    addr::A
+    S::NTuple{D,Int64}
+    aspect::NTuple{D,Float64}
+    vtable::Vector{Float64}  # interaction coefficients
+    g::Float64
+    impurity_only::Bool
+end
+
+function HOCartesianCentralImpurity(
+        addr::SingleComponentFockAddress; 
+        M::Int = num_modes(addr) - 1,
+        ηs = (), 
+        g = 1.0,
+        impurity_only = false
+    )
+    iseven(M) || throw(ArgumentError("M must be even"))
+    any(ηs .< 1.0) && throw(ArgumentError("Aspect ratios must be greater than 1.0"))
+    D = length(ηs) + 1
+    Srest = map(x -> 2floor(Int, M÷2 / x) + 1, ηs)
+    S = (M + 1, Srest...)
```

Comment:
Just a question here: Wouldn't it make sense to allow passing the tuple `S`, which as I understand is the number of modes per cartesian dimension in addition to automatically calculating it from `M`? E.g. if you wanted to scan through aspect ratios, it might be better to explicitly control (and maybe fix) the number of grid points per dimension.

Reply:
Only even parity states are affected so I felt it is better to enforce that rather than allow arbitrary `S` since it could produce unexpected result if not carefully checked. And it is easier to check and enforce this way.

If you want to scan aspect ratio use `ηs`

Reply:
The point is that the automatically calculated `S` will change at some point when you scan through `ηs`, which brings additional effects that may be difficult to dissociate from the change in the trap potential. 

> it could produce unexpected result if not carefully checked. 

Sure, but do we have to protect the user from any stupid mistake they could make? I have no objection to setting the default as it is right now, but I'd like to have the option of overriding the default and setting `S` independently from `ηs`.

Reply:
I added it back